gitbook/graphics-vertex-array.md

```diff
@@ -2,493 +2,440 @@
 
-To fill this gap, SFML provides a lower-level mechanism to draw things: Vertex arrays. As a matter of fact, vertex arrays are used internally by all other SFML classes. They allow for a more flexible definition of 2D entities, containing as many triangles as you need. They even allow drawing points or lines. 
+To fill this gap, CrSFML provides a lower-level mechanism to draw things: Vertex arrays. As a matter of fact, vertex arrays are used internally by all other CrSFML classes. They allow for a more flexible definition of 2D entities, containing as many triangles as you need. They even allow drawing points or lines.
 
 ## What is a vertex, and why are they always in arrays?
 
-A vertex is the smallest graphical entity that you can manipulate. In short, it is a graphical point: Naturally, it has a 2D position (x, y), but also a color, and a pair of texture coordinates. We'll go into the roles of these attributes later. 
+A vertex is the smallest graphical entity that you can manipulate. In short, it is a graphical point: Naturally, it has a 2D position (x, y), but also a color, and a pair of texture coordinates. We'll go into the roles of these attributes later.
 
-Vertices (plural of vertex) alone don't do much. They are always grouped into *primitives*: Points (1 vertex), lines (2 vertices), triangles (3 vertices) or quads (4 vertices). You can then combine multiple primitives together to create the final geometry of the entity. 
+Vertices (plural of vertex) alone don't do much. They are always grouped into *primitives*: Points (1 vertex), lines (2 vertices), triangles (3 vertices) or quads (4 vertices). You can then combine multiple primitives together to create the final geometry of the entity.
 
-Now you understand why we always talk about vertex arrays, and not vertices alone. 
+Now you understand why we always talk about vertex arrays, and not vertices alone.
 
 ## A simple vertex array
 
-Let's have a look at the [Vertex]({{book.api}}/Vertex.html) class now. It's simply a container which contains three public members and no functions besides its constructors. These constructors allow you to construct vertices from the set of attributes you care about -- you don't always need to color or texture your entity. 
+Let's have a look at the [Vertex]({{book.api}}/Vertex.html) class now. It's simply a container which contains three public members and no functions besides its constructors. These constructors allow you to construct vertices from the set of attributes you care about -- you don't always need to color or texture your entity.
 
-```
-// create a new vertex
-sf::Vertex vertex;
+```ruby
+# create a new vertex
+vertex = SF.vertex
 
-// set its position
-vertex.position = sf::Vector2f(10, 50);
+# set its position
+vertex.position = SF.vector2(10, 50)
```

Comment:
`field 'position' of struct CSFML::Vertex has type CSFML::Vector2f, not SF::Vector2(Float64)`